projects/Mallard/src/hooks/use-issue-summary.tsx

```diff
@@ -127,7 +127,16 @@ const IssueSummaryProvider = ({ children }: { children: React.ReactNode }) => {
             unsubNet()
             AppState.removeEventListener('change', appStateChangeListener)
         }
-    }, [issueId, maxAvailableEditions, issueSummary])
+    }, [issueId, issueSummary])
+
+    useEffect(() => {
+        ;(async () => {
+            if (maxAvailableEditions) {
+                const { isConnected } = await NetInfo.fetch()
+                grabIssueSummary(isConnected)
+            }
+        })()
+    }, [maxAvailableEditions])
```

Comment:
As mentioned offline, this second `useEffect` will cause two calls to `grabIssueSummary` concurrently and they'll compete with each other overriding the state. Not a blocker but we really should consolidate this eventually. Probably we should rely less on `useEffect` and have our state be in a separate, stateful class that does its own thing and subscribe to the right thing. Then, have the context provider just be a thin wrapper around that

Reply:
👍